src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -635,16 +635,27 @@ private BoundStatement BindUsingDeclarationStatementParts(LocalDeclarationStatem
         {
             Conversion iDisposableConversion;
             MethodSymbol disposeMethod;
+            bool hasAwait = node.AwaitKeyword != default;
             var declarations = BindUsingVariableDeclaration(
                                                 this,
                                                 diagnostics,
                                                 diagnostics.HasAnyErrors(),
                                                 node,
                                                 node.Declaration,
-                                                hasAwait: false,
+                                                hasAwait,
                                                 out iDisposableConversion,
                                                 out disposeMethod);
-            return new BoundUsingLocalDeclarations(node, disposeMethod, iDisposableConversion, declarations);
+
+            AwaitableInfo awaitOpt = null;
+            bool hasErrors = false;
+            if (hasAwait)
+            {
+                awaitOpt = BindAsyncDisposeAwaiter(node, node.AwaitKeyword, disposeMethod, diagnostics, ref hasErrors);
+            }
+
+            MessageID.IDS_FeatureUsingDeclarations.CheckFeatureAvailability(availableVersion: Compilation.LanguageVersion, diagnostics, node.Declaration.Location);
```

Comment:
>CheckFeatureAvailability [](start = 51, length = 24)

Feels like this should be a parsing diagnostic rather than a binding diagnostic, unless there is some difficulty I'm missing.

Reply:
Nope, i'm just still learning where things go :) will move to parsing.